include/zephyr/drivers/usb/udc_buf.h

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Buffers for USB device support
+ */
+
+#ifndef ZEPHYR_INCLUDE_UDC_BUF_H
+#define ZEPHYR_INCLUDE_UDC_BUF_H
+
+#include <zephyr/kernel.h>
+#include <zephyr/net/buf.h>
+
+#if defined(CONFIG_DCACHE) && !defined(CONFIG_UDC_BUF_FORCE_NOCACHE)
+/*
+ * Here we try to get DMA-safe buffers, but we lack a consistent source of
+ * information about data cache properties, such as line cache size, and a
+ * consistent source of information about what part of memory is DMA'able.
+ * For now, we simply assume that all available memory is DMA'able and use
+ * Kconfig option DCACHE_LINE_SIZE for alignment and granularity.
+ */
+#define Z_UDC_BUF_ALIGN		CONFIG_DCACHE_LINE_SIZE
+#define Z_UDC_BUF_GRANULARITY	CONFIG_DCACHE_LINE_SIZE
+#else
+/*
+ * Default alignment and granularity to pointer size if the platform does not
```

Comment:
Why the default is a pointer size, not simply 1 byte (any) alignment?

Reply:
DWC2 Buffer DMA needs 4 bytes, so using 1 byte here would break nRF54H20 USB with D-Cache disabled. Not sure how to properly describe the potential driver DMA alignment requirements.

I believe the buffers given out by default net buf allocator (this PR replaces them) are pointer size aligned.

Reply:
This is non-debatable default alignment for all USB controllers and has nothing to do with this PR.

Reply:
I understand some controllers have alignment requirements (like @tmon-nordic described for DWC2). However, this API seems to be controller-agnostic. That's why I was curious why controller-specific requirements are enforced here. That's OK if that's a common requirement for all supported controllers.

Anyway, it would be good to capture this decision and rationale in a comment to avoid misunderstandings in the future.

Reply:
Actually, this is good behavior to deliver the buffer with minimal alignment, this is what k_heap_alloc does. This prevents a lot of problems, not only with USB controllers. I will add a note about alignment requirements when I write documentation about buffer handling in USB.